lib/network/rest/server.js

```diff
@@ -0,0 +1,332 @@
+'use strict'; // eslint-disable-line
+
+const httpServer = require('../http/server');
+const Logger = require('werelogs').Logger;
+const url = require('url');
+const assert = require('assert');
+
+const constants = require('../../constants');
+const utils = require('./utils');
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into a two-integers array with possibly
+ * undefined values
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+ * @return {(Number|undefined) []} An array of [left, right] integer
+ *   values provided in the range specification if present, or
+ *   undefined otherwise.
+ */
+function parseRangeSpec(rangeHeader) {
+    const rangeMatch = /^bytes=([0-9]+)?-([0-9]+)?$/.exec(rangeHeader);
+    // ignore header if syntax is invalid
+    if (rangeMatch) {
+        return rangeMatch.slice(1, 3).map(
+            str => (str === undefined ?
+                    undefined : Number.parseInt(str, 10)));
+    }
+    return undefined;
```

Comment:
Instead of throwing and wrapping a try..catch around it, I find the pattern of returning an Error instance cleaner and clearer. 
```js
return new Error('invalid range header');
```
This would not throw.
You can add stack, message properties if you need them. Also, returning an error here would mean, you have to define the error only once and return it. While the other way would mean you are defining the error case multiple times.